docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+```bash
+if ! doguctl healthy --wait --timeout 120 postgresql; then
+  echo "timeout reached by waiting of ldap to get healthy"
+  exit 1
+fi
+```
+
+#### doguctl state
+
+Dieser Aufruft liest und schreibt Dogu-Zustandswerte. Er wird in Kombination mit einem [HealthCheck](../core/compendium_de.md#healthchecks) verwendet.
+
+```bash
+$ doguctl state "installing" # schreibt den Wert in den State
+# Erledige Installation
+
+$ doguctl state # Liest den Wert aus dem State
+installing
+
+$ doguctl state "ready" # Setzt den State auf den Standard-Healthy-Wert
+# Starte nun die Anwendung
+```
+
+#### doguctl random
+
+Dieser Aufruf erzeugt Zufallsstrings, geeignet um Passwörter oder sonstige Zugangsdaten zu erzeugen.
+
+```bash
+$ doguctl random # erzeugt Zufallsstring mit der Länge 16
+9HoF4nYmlLYtf6Ju
+$ doguctl random  -l 5 # erzeugt Zufallsstring mit gegebener Länge
+W6Wmj
+```
+
+#### doguctl template
+
+Dieser Aufruf erzeugt eine Datei aus einem [Golang-Template](https://pkg.go.dev/text/template). Unterschiedliche Registrywerte und Umgebungsvariablen können hierin direkt verarbeitet werden.
+
+Der Aufruf nimmt ein bis zwei Parameter entgegen:
+```bash
+$ doguctl template <Template-Datei> [Ausgabedatei]
+```
+
+Wird das Kommando `template` ohne Ausgabedatei verwendet, dann wird die gerenderte Ausgabe auf `stdout` ausgegeben.
+
+Unterstützte Template-Parameter:
+- Umgebungsvariablen
+  - `.Env.Get <Umgebungsvariable>` - verwendet vorher exportierte Umgebungsvariable "ADMIN_USERNAME"
+- Dogu-Konfiguration
+  - `.Config.Get <Dogu-Konfigurationsschlüssel>` - verwendet unverschlüsselten Dogu-Konfigurationswert unter `/config/<dogu>/<schlüssel>` 
+  - `.Config.GetOrDefault <Dogu-Konfigurationsschlüssel> <Fehlwert>` - verwendet unverschlüsselten Dogu-Konfigurationswert oder den bereitgestellten Fehlwert
+  - `.Config.GetAndDecrypt <verschlüsselter Dogu-Konfigurationsschlüssel>` - entschlüsselt den Dogu-Konfigurationswert und verwendet ihn
+  - `.Config.Exists <Dogu-Konfigurationsschlüssel>` - gibt einen `bool`-Wert zurück, ob ein Dogu-Konfigurationswert unter `/config/<dogu>/<schlüssel>` existiert
+- Globale Konfiguration
+  - `.GlobalConfig.Get <globaler Konfigurationsschlüssel>` - verwendet globalen Konfigurationswert `/config/_global/<schlüssel>`
+  - `.GlobalConfig.GetOrDefault <globaler Konfigurationsschlüssel> <Fehlwert>` - verwendet unverschlüsselten globalen Konfigurationswert oder den bereitgestellten Fehlwert
+  - `.GlobalConfig.Exists <Dogu-Konfigurationsschlüssel>` - gibt einen `bool`-Wert zurück, ob ein globaler Konfigurationswert unter `/config/_global/<schlüssel>` existiert
+- Dogu-Abfragen
+  - `.Dogus.IsEnabled <Dogu-Name>` - gibt einen `bool`-Wert zurück, ob ein Dogu installiert ist
+  - `.Dogus.Get <Dogu-Name>` - gibt ein [Dogu](../core/compendium_de.md#type-dogu)-Struct zurück
+  - `.Dogus.GetAll` - gibt ein Slice aller [Dogus](../core/compendium_de.md#type-dogu) zurück
```

Comment:
I don't know if everybody knows the meaning of slice if they didn't get in touch with Go before. Maybe add a hint for that?
```suggestion
  - `.Dogus.GetAll` - gibt ein Slice (eine Art Array in Go) aller [Dogus](../core/compendium_de.md#type-dogu) zurück
```

Reply:
Good point, but I do like to call a slice an array. Rather I'd should go about that the templating mechanism is Golang templating so then we can expect people to handle terms like this.

Reply:
Edit: In the very first paragraph we already tell people that this is a Golang template. I think this can stay this way.